app/javascript/components/explore/ExploreDisplayTabs.jsx

```diff
@@ -71,6 +72,19 @@ export default function ExploreDisplayTabs({
   const [isCellSelecting, setIsCellSelecting] = useState(false)
   // a plotly points_selected event
   const [currentPointsSelected, setCurrentPointsSelected] = useState(null)
+
+  const [showDeGroupPicker, setShowDeGroupPicker] = useState(false)
+  const [deGenes, setDeGenes] = useState(null)
+  const [deGroup, setDeGroup] = useState(null)
```

Comment:
You're totally right that "DE" requires a bit of context.  I opted for that because it's what's used by Ingest Pipeline and our product docs.  We've been talking in terms of the "DE MVP" and the "DE pipeline" and "inputs for DE" and have tentative API/model agreements for attributes like "has_DE".  For the unfamiliar reader, "diffEx" seems nearly as inscrutable as "DE": both really need to have "differential expression" spelled out in an adjacent comment for clarity.

I can't foresee maintainers confusing "DE" with e.g. German language assets, etc.  And given we've only used "DE" when abbreviating differential expression, introducing "diffExp" seems likely to introduce bugs at the seams of our different system components, confusion about where to use which abbreviation, etc.  One abbreviation is best.

If we change SCP's standard abbreviation for differential expression to not be "DE", I'd be inclined to use "diffEx" instead of "diffExp".  The former has unambiguous and simpler pronunciation, and is a tad shorter.  I'd also propose we use only one abbreviation (modulo language- and API-specific casing conventions) for this everywhere we abbreviate -- i.e. to swiftly rename related Ingest Pipeline modules and variables for clearer SCP-wide consistency.

I'd be really curious to get other reviewers' takes!

Reply:
Personally, I agree with the notion that DE is both internally consistent and no more or less inscrutable than diffEx.

Reply:
I don't see a strong need for the same abbreviations between docs and code -- docs often have plenty of context within--e.g. the start of a doc can say "Differential Expression (DE)" and then go from there, and so the need for the abbreviation to be self-explanatory is lessened. 

I see this as a particularly problematic case because we've chosen a very generic two-letter acronym that also isn't naturally read as an acronym.  we're reading it now as 'D.E.', but someone new to the codebase (or me in six months) will read it as 'deh' and have no idea--it's too easily lost in the rest of the variable names.  

You're right that I should have flagged this sooner in the ingest codebase--but it's also not as problematic there because it's not as interwoven into other code.  The places where `de` is used in ingest_pipeline.py are mostly in discrete sections led with headers like 

```elif "differential_expression" in arguments:```

which makes the context much cleaner

Reply:
Naming is really hard!  I'm glad we made space to discuss this [in Slack](https://broadinstitute.slack.com/archives/CBEHTH601/p1651673129447419?thread_ts=1651673125.085019&cid=CBEHTH601), and I think we reached consensus:
* Reduce how much we abbreviate "differential expression"
* When abbreviating / prefixing, only use "DE" (modulo case) like Ingest, Scanpy, Slack chats, prod docs, etc.  
 
With your other suggestions to do more to encapsulate the DE component code, I suspect there’ll also just plainly be fewer references to DE variables outside clearly labeled modules.  For example, while I've added a "Differential expression settings" comment here for clarity (743a1bf), I suspect those settings will be moved into a separate module like `DifferentialExpressionPanel` or somesuch in the upcoming MVP PR.